test/gtest/ucp/test_ucp_worker.cc

```diff
@@ -936,9 +984,10 @@ class test_worker_cpu_mask : public ucp_test {
     }
 
 protected:
+    // Don't create any entities on startup
     void init() override
     {
-        // Don't create any entities on startup
+        m_rlimit.set(RLIMIT_NOFILE, m_fd_per_cpu * ucs_sys_get_num_cpus());
```

Comment:
this cannot be pure raii in the test as we need `m_rlimit` to persist while `class test_worker_cpu_mask` is there.

Reply:
Maybe you can still have it with RAII by using `std::unique_ptr`?